Plugins/DD4hep/src/DD4hepBinningHelpers.cpp

```diff
@@ -0,0 +1,131 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Plugins/DD4hep/DD4hepBinningHelpers.hpp"
+
+void Acts::DD4hepBinningHelpers::decodeBinning(
+    dd4hep::rec::VariantParameters &variantParams, const xml_comp_t &xmlBinning,
+    const std::string &bname, const std::vector<std::string> &bvals) {
+  // Set the surface binninng parameter to true
+  variantParams.set<int>(std::string(bname + "_dim"), bvals.size());
+  for (const auto &bv : bvals) {
+    // Gather the number of bins, 0 indicates variable binning
+    int nBins = Acts::getAttrValueOr<int>(xmlBinning, std::string("n" + bv), 0);
+    // Gather the bin expansion parameter, expansion of 0 is default
+    int nExpansion =
+        Acts::getAttrValueOr<int>(xmlBinning, std::string(bv + "expansion"), 0);
+    // Auto-range detection
+    bool autoRange = Acts::getAttrValueOr<bool>(
+        xmlBinning, std::string(bv + "autorange"), false);
+    variantParams.set<bool>(bname + "_" + bv + "_autorange", autoRange);
+    variantParams.set<int>(bname + "_" + bv + "_exp", nExpansion);
+    // Equidistant binning detected
+    if (nBins > 0) {
```

Comment:
`nBins == 0` means variable binning? How are the number of variable bins defined? Explicitly from the boundaries it seems?

It might be a bit confusing that `nBins>0` and boundaries are mutually exclusive.

Reply:
This code is only used in the UnitTests now, it only serves the local testing, so I think I will leave it as such.